### PR TITLE
Publish documentation snapshot only if necessary

### PR DESCRIPTION
**Pull request by sormuras**
## Overview

Prior to this commit the current documentation sources were always compiled, built, prepared, and published even if nothing was edited.

Now, the file `published-checksum.txt` containing a checksum for each documentation source asset is used to determine if a build and publish is necessary. It will show up here: https://github.com/junit-team/junit5/tree/gh-pages/docs/snapshot

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

### CODE REVIEW

File: src/publishDocumentationSnapshotOnlyIfNecessary.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  # no changes detected: we're done
+  #
+  echo "Already published documentation with same source checksum."
+else
+  #
+  # update checksum file and trigger new documentation build and upload
+  #
+  echo "Creating and publishing documentation..."
+  cp --force "${current}" "${published}"
+  echo "./gradlew --scan gitPublishPush"
```
Comment (sormuras) [inline src/publishDocumentationSnapshotOnlyIfNecessary.sh]: Dooh. Why is "poor man's dry-run mode" still active?!

**Comment by sormuras** [comment]
Including all `src/main/java/**.java` files to detect potential changes in Javadoc comments.

File: src/publishDocumentationSnapshotOnlyIfNecessary.sh [hunk 2]
```diff
@@ -26,6 +29,6 @@ else
   # update checksum file and trigger new documentation build and upload
   #
   echo "Creating and publishing documentation..."
-  cp --force "${current}" "${published}"
-  echo "./gradlew --scan gitPublishPush"
+  cp --force "${current}" "${published}"ca
```
Comment (marcphilipp) [inline src/publishDocumentationSnapshotOnlyIfNecessary.sh]: `ca`?

Comment (marcphilipp) [reply src/publishDocumentationSnapshotOnlyIfNecessary.sh]: Why so forcefully?

Comment (sormuras) [reply src/publishDocumentationSnapshotOnlyIfNecessary.sh]: `ca` ... _ceine ahnung_ ;-)

`--force` to overwrite existing one.

**Comment by sormuras** [comment]
The script-based guard prevents any Gradle task to start at all. See how many CPU cycles are "wasted" for nothing: https://scans.gradle.com/s/sdegvcf4c4sra/performance/execution

**Pull request merged**